src/Evolution/Initialization/Interface.hpp

```diff
@@ -41,67 +51,76 @@ namespace Initialization {
-      Tags::InterfaceComputeItem<Tags::BoundaryDirectionsExterior<dim>,
-                                 Tags::InterfaceMesh<dim>>,
-      Tags::Slice<Tags::BoundaryDirectionsExterior<dim>,
-                  typename System::spacetime_variables_tag>,
-      Tags::Slice<Tags::BoundaryDirectionsExterior<dim>,
-                  typename System::primitive_variables_tag>,
-      Tags::InterfaceComputeItem<Tags::BoundaryDirectionsExterior<dim>,
-                                 Tags::BoundaryCoordinates<dim>>,
-      Tags::InterfaceComputeItem<Tags::BoundaryDirectionsExterior<dim>,
-                                 Tags::UnnormalizedFaceNormal<dim>>,
-      Tags::InterfaceComputeItem<Tags::BoundaryDirectionsExterior<dim>,
-                                 typename System::template magnitude_tag<
-                                     Tags::UnnormalizedFaceNormal<dim>>>,
-      Tags::InterfaceComputeItem<
-          Tags::BoundaryDirectionsExterior<dim>,
-          Tags::NormalizedCompute<Tags::UnnormalizedFaceNormal<dim>>>>;
+      ::Tags::BoundaryDirectionsExterior<dim>,
+      ::Tags::InterfaceComputeItem<::Tags::BoundaryDirectionsExterior<dim>,
+                                   ::Tags::Direction<dim>>,
+      ::Tags::InterfaceComputeItem<::Tags::BoundaryDirectionsExterior<dim>,
+                                   ::Tags::InterfaceMesh<dim>>,
+      ::Tags::Slice<::Tags::BoundaryDirectionsExterior<dim>,
+                    typename System::spacetime_variables_tag>,
+      ::Tags::Slice<::Tags::BoundaryDirectionsExterior<dim>,
+                    typename System::primitive_variables_tag>,
+      ::Tags::InterfaceComputeItem<::Tags::BoundaryDirectionsExterior<dim>,
+                                   ::Tags::BoundaryCoordinates<dim>>,
+      ::Tags::InterfaceComputeItem<::Tags::BoundaryDirectionsExterior<dim>,
+                                   ::Tags::UnnormalizedFaceNormal<dim>>,
+      ::Tags::InterfaceComputeItem<::Tags::BoundaryDirectionsExterior<dim>,
+                                   typename System::template magnitude_tag<
+                                       ::Tags::UnnormalizedFaceNormal<dim>>>,
+      ::Tags::InterfaceComputeItem<
+          ::Tags::BoundaryDirectionsExterior<dim>,
+          ::Tags::NormalizedCompute<::Tags::UnnormalizedFaceNormal<dim>>>>;
 
   using compute_tags =
-      tmpl::append<face_tags<Tags::InternalDirections<dim>>,
-                   face_tags<Tags::BoundaryDirectionsInterior<dim>>, ext_tags>;
+      tmpl::append<face_tags<::Tags::InternalDirections<dim>>,
+                   face_tags<::Tags::BoundaryDirectionsInterior<dim>>,
+                   ext_tags>;
 
-  template <typename TagsList>
-  static auto initialize(db::DataBox<TagsList>&& box) noexcept {
-    const auto& mesh = db::get<Tags::Mesh<dim>>(box);
+  template <typename DbTagsList, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent>
```

Comment:
Check for dependencies

Reply:
There are none...

Reply:
This needs domain things. Now I'm confused, how does this compile when it can't be sure `Tags::Mesh` is in the DataBox? Why don't you have to add a `Requires` so that only when the incoming DataBox contains the dependencies will this `apply` run? `merge_into_databox` doesn't protect against missing dependencies, or did I miss something?
Unresolving this comment just so that I understand what's going on.

Reply:
This doesn't depend on any of the initialization options. This action behaves the same as pretty much every other action in the code.

Reply:
It's in the init phase though, so it's DataBox has to work with _all_ DataBoxes that come out of the other actions in this phase. Does this work because the domain initializer is the _first_ action in the list, so the Interface initializer is guaranteed to have the domain stuff in the DataBox?

Reply:
Oh, I see your confusion now. Yes, because you cannot have interfaces without a domain, the domain is initialized first and so this works. Does that make sense?

Reply:
Yes. If the domain was initialised, say, second in the init action list, then I think we would get compiler errors because it's not guaranteed that the actions are run in order.

Reply:
So I think the initialisers have to check their dependencies. We can't rely on the domain initializer always being the very first in the list.

Reply:
Maybe I am still getting something wrong, actually. The interface initializer slices variables to the faces, but how can it be sure the system initializer was run? In principle the algorithm could jump from the domain action directly to the interface action..

Reply:
My personal view here is that at some point we need users to think about what they're doing. Trying to safeguard against everything is impossible, and will drive compile time through the roof. It's these types of situations where I think we should use documentation (which I'm happy to add here) and examples.